main/javascript-modifications/ses-guide.md

```diff
@@ -0,0 +1,506 @@
+that inherit from it. `harden()` does transitive freezing by following the object’s own
+properties (as opposed to properties it inherited), and the objects whose own properties refer 
+to them, and so forth.
+
+When you freeze an object via `harden()`, it ensures any external callers 
+can only interact with it through functions in the object’s API. `harden()` 
+is an enhanced transitive version of `Object.freeze()`, which only locks up an 
+object's own properties.
+
+All objects that are transferred across a trust boundary must have their API
+surface frozen, usually by calling `harden()`. This ensures other objects can 
+only interact with them through their defined method interface. *CapTP*, our 
+communications layer for passing references to distributed objects, enforces 
+this at vat boundaries.
+
+The general rule is that if you make a new object and give it to someone else 
+(and don't immediately forget it yourself), you should give them `harden(obj)` 
+instead of the raw object. This prevents someone from adding/deleting the properties
+or prototypes of that object. Being hardened doesn't preclude an object from having 
+access to mutable state (`harden(new Map())` still behaves like a normal mutable `Map`), 
+but it means their methods stay the same and can't be surprisingly changed by someone else
+
+Defined objects (`mint`, `issuer`, `zcf`, etc.) shouldn't need hardening as their 
+constructors should do that work. It's mainly records, callbacks, and ephemeral 
+objects that need hardening.
+
+You can send a message to a hardened object. If it's a record, you can access 
+its properties and their values.
+
+`harden()` is automatically provided by SES. Any code that will run inside a vat or a 
+contract can use harden as a global, without importing anything.
+
+Tip: If your text editor/IDE complains about `harden()` not being defined or imported, 
+try adding `/* global harden */` to the top of the file.
+
+You use `harden()` like this:
+```js
+const o = {a: 2};
+o.a  = 12;
+console.log(o.a); // 12 because o is still mutable
+harden(o);
+o.a  = 37; // throws a TypeError because o is now hardened
+```
+## `lockdown()` and `harden()`
+
+`lockdown()` and `harden()` essentially do the same thing; freeze objects so their 
+properties cannot be changed. The only way to interact with frozen objects is through 
+their methods. Their differences are what objects you use them on, and when you use them.
+
+`lockdown()` **must** be called first. It hardens JavaScript's built-in *primordials* 
+(implicitly shared global objects) and enables `harden()`. If you call `harden()` 
+before `lockdown()` excutes, it throws an error.
+
+`lockdown()` works on objects created by the JavaScript language itself as part of 
+its definition. Use `harden()` to freeze objects created after `lockdown()`was called;
+i.e. objects created by programs written in JavaScript.
```

Comment:
This appears to be a duplicate in this document.

Reply:
Not sure what you mean? Some information's repeated here to make clear what the differences between lockdown and harden, but just enough to make clear how they are different and when to use one or the other?